pyvista/core/filters/data_set.py

```diff
@@ -5850,6 +5850,237 @@ def extract_cells_by_type(self, cell_types, progress_bar=False):
+        progress_bar=False,
+        inplace=False,
+    ):
+        """Renumber labeled data such that labels are contiguous.
+
+        This filter renumbers scalar label data of any type with ``N`` labels
+        such that the output labels are contiguous from ``[0, N)``. The
+        output may optionally be sorted by label count.
+
+        See also :func:`pyvista.DatasetFilters.sort_labels`.
+
+        Notes
+        -----
+        This filter uses ``vtkPackLabels`` as the underlying method which
+        requires VTK version 9.3 or higher. If ``vtkPackLabels`` is not
+        available, packing is done with ``NumPy`` instead which may be
+        slower. For best performance, consider upgrading VTK.
+
+        Parameters
+        ----------
+        sort : bool, default: False
+            Whether to sort the output by label count in descending order
+            (i.e. from largest to smallest).
+
+        scalars : str, optional
+            Name of scalars to pack. Defaults to currently active scalars.
+
+        preference : str, default: "point"
+            When ``scalars`` is specified, this is the preferred array
+            type to search for in the dataset.  Must be either
+            ``'point'`` or ``'cell'``.
+
+        output_scalars : str, None
+            Name of the packed output scalars. By default, the output is
+            saved to currently active scalars.
+
+        progress_bar : bool, default: False
+            If ``True``, display a progress bar. Has no effect if VTK
+            version is lower than 9.3.
+
+        inplace : bool, default: False
+            If ``True``, the mesh is updated in-place.
+
+        Returns
+        -------
+        pyvista.Dataset
+            Dataset with packed labels.
+
+        Examples
+        --------
+        Pack segmented image labels.
+
+        >>> # Load non-contiguous image labels
+        >>> from pyvista import examples
+        >>> import numpy as np
+        >>> image_labels = examples.download_frog_tissue()
+        >>>
+        >>> # Show range of labels
+        >>> image_labels.get_data_range()
+        (0, 29)
+        >>>
+        >>> # Find 'gaps' in the labels
+        >>> label_numbers = np.unique(image_labels.active_scalars)
+        >>> label_max = np.max(label_numbers)
+        >>> missing_labels = set(range(label_max)) - set(label_numbers)
+        >>> len(missing_labels)
+        4
+        >>>
+        >>> # Pack labels to remove gaps
+        >>> packed_labels = image_labels.pack_labels()
+        >>>
+        >>> # Show range of packed labels
+        >>> packed_labels.get_data_range()
+        (0, 25)
+
+        """
+        # Determine input scalars
+        if scalars is None:
+            set_default_active_scalars(self)
+            _, scalars = self.active_scalars_info
+        arr = get_array(self, scalars, preference=preference, err=False)
+        if arr is None:
+            raise ValueError('No arrays present to sort.')
```

Comment:
In my mind, I vote to leave error handling to get_array to reduce maintenance costs, and I don't think users will have much trouble since Python shows which function caused the error when it occurs.
```suggestion
        arr = get_array(self, scalars, preference=preference, err=True)
```

Reply:
Agreed. This code block was borrowed from `threshold`. Should `threshold` have this check removed too? Also, `test_sort_labels` has a test case written specifically to check that `ValueError` is raised with no data. Should this be removed as well, since that error should be tested / handled by `get_array`?

Reply:
Yes. Please. I guess we should change it too.